app/src/main/java/com/klk/dailyjournal/SecondActivity.kt

```diff
@@ -0,0 +1,227 @@
+package com.klk.dailyjournal
+
+import android.Manifest
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.graphics.Color
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.os.Environment
+import android.provider.MediaStore
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import android.widget.Toast
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityCompat
+import androidx.core.content.FileProvider
+import com.klk.dailyjournal.data.NoteEntity
+import com.klk.dailyjournal.data.NoteRepository
+import com.klk.dailyjournal.entities.Feeling
+import java.io.File
+import java.io.FileOutputStream
+import java.text.SimpleDateFormat
+import java.util.*
+
+
+class SecondActivity: AppCompatActivity(){
+    val calendar = Calendar.getInstance()
+
+    var photo: File? = null
+    var pathPhoto: String? = null
+    var  mood: Feeling?= null
+    val repo = NoteRepository.get()
+
+
+    val TAG = "xyz"
+    private val TAKE_PHOTO_PERMISSION_REQUEST_CODE = 1
+
+
+    fun saveNote(view: View){
+        val gratefulnessEdit = findViewById<TextView>(R.id.gratefulness_edit)
+        val todayEdit = findViewById<TextView>(R.id.today_edit)
+        val myNoteEdit = findViewById<TextView>(R.id.my_note_edit)
+
+        val path  = if (pathPhoto==null)  photo?.path else pathPhoto
+        repo.insert(NoteEntity(
+            0,
+            SimpleDateFormat("EEEE", Locale.ENGLISH).
+            format(getDate()),
+            "${getDayAsString()} ${getMonthName()}",
+            "face_temporary.png",
+            gratefulnessEdit.text.toString(),
+            todayEdit.text.toString(),
+            myNoteEdit.text.toString(),
+            path, null))
+
+            //save
```

Comment:
when we click done we should be moved back to main activity, also do we allow to add empty data?